### ISSUE TITLE
Error: file not found when using Docker registry

### ISSUE THREAD

**Issue by cmdrrobin** [minio/minio#6558]
When using Docker registry to upload Docker images I get a 403 error message.

`error parsing HTTP 403 response body: unexpected end of JSON input: ""`

I have tested it with the following versions:
- Docker registry: 2.6, 2.5, 2.5.1, 2.5.0 and latest
- Minio: RELEASE.2018-09-12T18-49-56Z, latest and edge.

This is a follow-up on issue https://github.com/minio/minio/issues/5902

## Expected Behavior
Expected behaviour is that the Docker image is uploaded sucessfully.

## Current Behavior
```
The push refers to repository [hub.local/alpine]
cd7100a72410: Pushing [==================================================>]  4.403MB
error parsing HTTP 403 response body: unexpected end of JSON input: ""
```

The following message is shown in Minio:
```
API: SYSTEM()
Time: 08:39:46 UTC 10/03/2018
Error: file not found
       1: cmd/logger/logger.go:294:logger.LogIf()
       2: cmd/fs-v1.go:1029:cmd.(*FSObjects).listDirFactory.func1()
       3: cmd/tree-walk.go:145:cmd.doTreeWalk()
       4: cmd/tree-walk.go:253:cmd.startTreeWalk.func1()
```

## Possible Solution
No fix or workaround founded

## Steps to Reproduce (for bugs)
The following registry config:
```
version: 0.1
log:
    fields:
        service: registry
http:
    addr: :5000
    tls:
      certificate: /etc/docker/registry/certs/server.crt
      key: /etc/docker/registry/certs/server.key
storage:
    cache:
        layerinfo: inmemory
    s3:
        accesskey: <<removed>>
        secretkey: <<removed>>
        region: eu-west-1
        regionendpoint: http://minio:9000
        bucket: docker
        encrypt: false
        secure: false
        v4auth: true
        chunksize: 5242880
        rootdirectory: /
```

Create a bucket called `docker` with policy "public"

```
$ ./mc policy public local/docker
Access permission for `local/docker` is set to `public`
```

## Context
Both Docker registry and Minio running on same node.

## Your Environment
OS: RancherOS v1.3.0

Docker version
```
Client:
 Version:      17.03.2-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   f5ec1e2
 Built:        Tue Jun 27 01:35:00 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.2-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   f5ec1e2
 Built:        Tue Jun 27 01:35:00 2017
 OS/Arch:      linux/amd64
 Experimental: false
```
Run commands:
`docker run -d --name registry -p 443:5000 -v /mnt/data/registry/conf:/etc/docker/registry:ro --network webservice registry:latest`
`docker run -d --rm --network webservice --name minio -v /mnt/data/minio:/data minio/minio server /data`

Minio version:
```
Version: 2018-09-25T21:34:43Z
Release-Tag: RELEASE.2018-09-25T21-34-43Z
Commit-ID: aa4e2b1542b98097e08680f21b790de0b776378c
```

**Comment by kannappanr** [comment]
@rgruyters Thanks for filing this issue. We will try to reproduce the issue and get back to you.

**Comment by kannappanr** [comment]
@rgruyters We tried this on an `ubuntu 17.10` machine with `docker version`

```
Client:
 Version:           18.06.0-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        0ffa825
 Built:             Wed Jul 18 19:09:56 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.0-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       0ffa825
  Built:            Wed Jul 18 19:07:55 2018
  OS/Arch:          linux/amd64
  Experimental:     false
```
Minio version tested was the latest master and i don't think that is an issue here. 

The contents of `config.yml` used was 
```
version: 0.1
log:
    fields:
        service: registry
http:
    addr: :5000
#    tls:
#      certificate: /etc/docker/registry/certs/server.crt
#      key: /etc/docker/registry/certs/server.key
storage:
    cache:
        layerinfo: inmemory
    s3:
        accesskey: minio
        secretkey: minio123
        region: us-east-1
        regionendpoint: http://192.168.1.65:9000
        bucket: docker
        encrypt: false
        secure: false
        v4auth: true
        chunksize: 5242880
        rootdirectory: /
```

As far as minio is concerned, I issued the following commands
`mc mb myminio/docker`
`mc policy public local/docker`
 Here are the commands that I issued after that
`docker run -d --name registry -p 5000:5000 -v /tmp/conf:/etc/docker/registry:ro registry:latest`
`docker image tag minio/minio localhost:5000/testimage`
`docker push localhost:5000/testimage`

The push succeeded for me.  As far as the error seen on the console, I am also getting it. I don't think it matters. Is it possible for you to test it on a different version of `docker` and maybe also `Ubuntu`

**Comment by cmdrrobin** [comment]
Found the problem. Our environment has a upstream proxy configured. The minio hostname was not excluded from proxy settings.
After changing this I found another issue.

```
The push refers to repository [hub.local/alpine]
cd7100a72410: Pushing  4.403MB
dial tcp: lookup minio: no such host
```

It looks like after the image has been pushed the local Docker engine wants to access Minio directly. Is this really needed, because we have Minio deployed behind the Docker Registry.
